backend/app/services/pdf_reader.py

```diff
@@ -1,22 +1,52 @@
 from typing import BinaryIO
 from dataclasses import dataclass
 import pymupdf
+import pytesseract
 from io import StringIO
-
-# TODO Marta / Luca: I let you make this better. The signature of the function should be unchanged.
+import re
+from pdf2image import convert_from_bytes
 
 @dataclass
 class PDFReader:
+    file_type: str = "pdf"
     sort: bool = True
+    format_type: str = 'words'
```

Comment:
Why that?

I think this class should be used as a simple [curried](https://en.wikipedia.org/wiki/Currying) function where an instance is constructed with config attributes (not changing often or changing with the config of the user, and common to all methods) and the arguments of `as_text` the actual arguments changing from one call to the other.

Another possible use-case of the attributes is when the function changes some state, but it's not the case here.